Create rucio.cfg from inside Rucio objectstore

SergeyYakubov opened the pull request:
Rucio client library requires rucio.cfg to be present on the system. We create this file now in objectstore to make sure it always gets created.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

SergeyYakubov committed:
create rucio.cfg from inside objectstore

nsoranzo commented on lib/galaxy/objectstore/rucio.py:
```diff
@@ -293,7 +311,7 @@ def __init__(self, config, config_dict):
 
         self.oidc_provider = config_dict.get("oidc_provider", None)
-        self.rucio_broker = RucioBroker(self.rucio_config)
+        self.rucio_broker = RucioBroker(self.rucio_config, self.config)
```
```suggestion
        self.rucio_broker = RucioBroker(self)
```

nsoranzo commented on lib/galaxy/objectstore/rucio.py:
```diff
@@ -138,9 +138,11 @@ def parse_config_xml(config_xml):
         self._temp_file_name = None
+        self.rucio_config_created = False
         self.config = rucio_config
+        self.extra_dirs = config.get("extra_dirs", [])
```
```suggestion
        self.config = rucio_object_store.rucio_config
        self.extra_dirs = rucio_object_store.extra_dirs
```

nsoranzo commented on lib/galaxy/objectstore/rucio.py:
```diff
@@ -151,12 +153,28 @@ def __init__(self, rucio_config):
+            temp_directory = "/tmp"
+            for extra_dir in self.extra_dirs:
+                if extra_dir["type"] == "temp":
+                    temp_directory = extra_dir["path"]
```
```suggestion
            temp_directory = self.extra_dirs["temp"]
```

nsoranzo reviewed (commented):
Thanks for the fix! I think you can rely on the objectstore `extra_dirs["temp"]` as defined in `lib/galaxy/objectstore/__init__.py` , see suggestions.

SergeyYakubov committed:
refactor rucio broker config

mvdbeek committed:
Set RUCIO_CONFIG every time client is used

mvdbeek committed:
Write config in one pass

mvdbeek committed:
Disable celery and extended metadata

Either of these options are not compatible with rucio. Maybe more than
one connection isn't permitted ?

mvdbeek committed:
Don't collect twice

mvdbeek committed:
Run in extended metadata mode again

mvdbeek reviewed (approved):
The failing test is unrelated
